interfaces/builtin/mir_test.go

```diff
@@ -51,24 +53,61 @@ var _ = Suite(&MirInterfaceSuite{
 	},
 })
 
+func (s *MirInterfaceSuite) SetUpTest(c *C) {
+	s.slot = &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "mir-server", Type: snap.TypeOS},
+			Name:      "mir-server",
+			Interface: "mir",
+		},
+	}
+}
+
 func (s *MirInterfaceSuite) TestName(c *C) {
 	c.Assert(s.iface.Name(), Equals, "mir")
 }
 
 func (s *MirInterfaceSuite) TestUsedSecuritySystems(c *C) {
-	systems := [...]interfaces.SecuritySystem{interfaces.SecurityAppArmor,
-		interfaces.SecuritySecComp}
-	for _, system := range systems {
-		snippet, err := s.iface.PermanentSlotSnippet(s.slot, system)
-		c.Assert(err, IsNil)
-		c.Assert(snippet, Not(IsNil))
-		if system != interfaces.SecuritySecComp {
-			snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, system)
-			c.Assert(err, IsNil)
-			c.Assert(snippet, Not(IsNil))
-			snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, system)
-			c.Assert(err, IsNil)
-			c.Assert(snippet, Not(IsNil))
-		}
+	snippet, err := s.iface.PermanentSlotSnippet(s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+	snippet, err = s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+	snippet, err = s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+}
+
+func (s *MirInterfaceSuite) TestSecComp(c *C) {
+	s.slot = &interfaces.Slot{
+		SlotInfo: &snap.SlotInfo{
+			Snap:      &snap.Info{SuggestedName: "mir-server"},
+			Name:      "mir-server",
+			Interface: "mir",
+			Apps: map[string]*snap.AppInfo{
+				"mir": {
+					Snap: &snap.Info{
+						SuggestedName: "mir-server",
+					},
+					Name: "mir"}},
+		},
 	}
+
+	seccompSpec := &seccomp.Specification{}
+	err := seccompSpec.AddPermanentSlot(s.iface, s.slot)
+	c.Assert(err, IsNil)
+	snippets := seccompSpec.Snippets()
+	c.Assert(len(snippets), Equals, 1)
+	c.Assert(len(snippets["snap.mir-server.mir"]), Equals, 1)
+	c.Check(string(snippets["snap.mir-server.mir"][0]), testutil.Contains, "listen\n")
+}
+
+func (s *MirInterfaceSuite) TestSecCompOnClassic(c *C) {
```

Comment:
I don't understand this test, it says "Classic" in the name but I see nothing related to classic.